### PR TITLE
phpunit 6 update

### PR DESCRIPTION
**Pull request by kokspflanze**
- phpunit 4 droped
- added phpunit 5.7.14 and phpunit 6.0.7
- added ::class syntax in tests

### CODE REVIEW

**Comment by kokspflanze** [comment]
the php7 latest part should be fixed with https://github.com/zendframework/zend-eventmanager/pull/47

File: test/Application/BadControllerTrait.php [hunk 1]
```diff
@@ -72,7 +73,7 @@ public function prepareApplication()
                     },
                 ],
                 'invokables' => [
-                    'Request'              => 'Zend\Http\PhpEnvironment\Request',
+                    'Request'              => Request::class,
```
Comment (froschdesign) [inline test/Application/BadControllerTrait.php:76]: The topic of this PR is "PHPUnit 6" and all these changes makes a review more difficult.

Comment (kokspflanze) [reply test/Application/BadControllerTrait.php:76]: if you want i can try to split it in 2 PRs

File: .travis.yml [hunk 2]
```diff
@@ -74,6 +72,7 @@ install:
   - if [[ $DEPS == 'latest' ]]; then travis_retry composer update $COMPOSER_ARGS ; fi
   - if [[ $DEPS == 'lowest' ]]; then travis_retry composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
   - if [[ $TEST_COVERAGE == 'true' ]]; then travis_retry composer require --dev $COMPOSER_ARGS satooshi/php-coveralls:^1.0 ; fi
+  - if [[ $TRAVIS_PHP_VERSION =~ ^5.6 ]]; then travis_retry composer update $COMPOSER_ARGS --with-dependencies $LEGACY_DEPS ; fi
```
Comment (michalbundyra) [inline .travis.yml]: Order here is important. This is not gonna work as expected if we will have composer.lock locked with PHPUnit 6.0.

I would suggest also extract `satooshi/php-coveralls` to variable, as we have it in other modules (please see for example some expressive modules).

Comment (kokspflanze) [reply .travis.yml]: i can change it to https://github.com/zendframework/zend-expressive-fastroute/blob/master/.travis.yml if that is okay

Comment (michalbundyra) [reply .travis.yml]: Yes, it is right. Please update. Thanks!

**Pull request merged**